lsbi/network.py

```diff
@@ -6,47 +6,57 @@
 
 
 class BinaryClassifierBase(nn.Module):
-    """Base model for binary classification. Following 2305.11241."""
+    """Base model for binary classification. Following 2305.11241.
+
+    A simple binary classifier:
+    - 5 hidden layers
+        - Layer 1 with 130 units
+        - Layers 2-4 with 16 units
+    - Leaky ReLU activation function
+    - Batch normalization
+    - Output layer with 1 unit linear classifier unit
+    - Adam optimizer with learning rate 0.001
+    - Exponential learning rate decay with decay rate 0.95
+    - Binary cross entropy loss function
+
+    and 16 hidden units per layer.
+
+    Parameters
+    ----------
+    input_dim : int
+        Dimension of the input data.
+    internal_dim : int, optional (default=16)
+        Dimension of the internal layers of the network.
+    initial_dim : int, optional (default=130)
+        Dimension of the first layer of the network.
+    """
 
     def __init__(self, input_dim, internal_dim=16, initial_dim=130):
         super(BinaryClassifierBase, self).__init__()
 
-        self.dense = nn.Linear(input_dim, initial_dim)
-        self.leaky_relu = nn.LeakyReLU()
-        self.batch_norm = nn.BatchNorm1d(initial_dim)
-        self.dense_1 = nn.Linear(initial_dim, internal_dim)
-        self.leaky_relu_1 = nn.LeakyReLU()
-        self.batch_norm_1 = nn.BatchNorm1d(internal_dim)
-        self.dense_2 = nn.Linear(internal_dim, internal_dim)
-        self.leaky_relu_2 = nn.LeakyReLU()
-        self.batch_norm_2 = nn.BatchNorm1d(internal_dim)
-        self.dense_3 = nn.Linear(internal_dim, internal_dim)
-        self.leaky_relu_3 = nn.LeakyReLU()
-        self.batch_norm_3 = nn.BatchNorm1d(internal_dim)
-        self.dense_4 = nn.Linear(internal_dim, internal_dim)
-        self.leaky_relu_4 = nn.LeakyReLU()
-        self.dense_5 = nn.Linear(internal_dim, 1)
+        self.model = nn.Sequential(
```

Comment:
This felt more 'torchy', and less error prone. It does remove the named batch_norm attributes, which I couldn't quite work out why they were needed.

Reply:
Yes much nicer! The batch norm attributes don't need to be individually exposed anyway